l10n_it_vat_statement_communication/migrations/16.0.1.1.0/pre-migrate.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from openupgradelib import openupgrade
+
+_logger = logging.getLogger(__name__)
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    if not version:
+        return
+
+    rm_old_vat_communication_multi_company_rule(env)
+
+
+def rm_old_vat_communication_multi_company_rule(env):
+    _logger.info(
+        "Migration of l10n_it_vat_statement_communication - search for"
+        " previous multi company rule for 'comunicazione.liquidazione' model"
+    )
+    module = "l10n_it_vat_statement_communication"
+    rule_xml_id = "comunucazione_liquidazione_iva_multi_company"
+    old_vat_comm_multi_company_rule_ref = env.ref(
+        f"{module}.{rule_xml_id}", raise_if_not_found=False
+    )
+
+    if not old_vat_comm_multi_company_rule_ref:
+        # avoid unlink not existent record
+        _logger.warning(
+            "Migration of l10n_it_vat_statement_communication - previous"
+            " multi company rule for 'comunicazione.liquidazione' model not"
+            " found"
+        )
+        return
+
+    _logger.info(
+        "Migration of l10n_it_vat_statement_communication - unlink"
+        " previous multi company rule for 'comunicazione.liquidazione' model"
+    )
+    old_vat_comm_multi_company_rule_ref.unlink()
```

Comment:
Non credo che ricreare la regola sia _safe_: l'utente potrebbe averla modificata (in fondo è per quello che è `noupdate`), oppure se fosse collegata ad altri record tramite campi relazionali il collegamento sparirebbe.

Normalmente quando una migrazione deve modificare dei record `noupdate` è sufficiente aggiornare il record, in questo caso però non si sta modificando il record ma il suo `ir.model.data`, quindi la migrazione dovrebbe recuperare l'`ir.model.data` e modificarlo.

Reply:
No, lo script di migrazione mira a ricreare la regola, che in ambienti migrati resta alla vecchia versione https://github.com/OCA/l10n-italy/blob/12.0/l10n_it_vat_statement_communication/security/security.xml#L9 causando problemi con ambienti multicompany perché il campo usato nella rule potrebbe non corrispondere al valore della proprietà company nell'env che da 14.0 viene usato per molti dei casi; il modificare ir_model_data è solo una correzione ortografica, `comunucazione` non lo trovo corretto